charts/nexus-iq/templates/deployment.yaml

```diff
@@ -54,7 +56,7 @@ spec:
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
           volumeMounts:
-            - mountPath: /sonatype-work
+            - mountPath: {{ .Values.configYaml.sonatypeWork }}
```

Comment:
Now any reference to the `sonatype-work` strings in the `deployment.yaml` template comes from `{{ .Values.configYaml.sonatypeWork }}`

Reply:
now i'm kind of wondering why a user would want to change `/sonatype-work`. We mount whatever `nxiq-pv-data` volume they want there. it seems it may have been sufficient to hardcode the instance of `${SONATYPE_WORK}` to `/sonatype-work` _or_ merely set the `env:` of `SONATYPE_WORK` to `/sonatype-work` and then it's done.  I _think_ a user's potential point of customization should be choosing the volumes to mount at `/sonatype-work`, `/var/log/nexus-iq-server`, `/etc/nexus-iq-server`, and `/etc/nexus-iq-license`. Of course, we allow the user to change `secretMountName` below, so I guess there's a precedent for some reason.

Reply:
Since it needs to be in that `configYaml:` part of `values.yaml`, and someone _could_ change it there, I guess we need to facilitate it and use the value provided there.

Reply:
yes, I decided to go this way to reduce the options of having something wrong in the configuration of the sonatype work directory. So, now if the user decides to change `configYaml` on of `values.yaml` the change will be reflected to the volumes and the `JAVA_OPTS` env variable. At least with this, the configuration will be consistent.

For other configurations like the ones for the logs and some other volumes mount paths I think we can externalize those as new configuration options, but I would do that in a different ticket. As you said we are already allowing that for `secretMountName`, so I think that may work.